### PR TITLE
calendar-adjustment

### PR DESCRIPTION
**Pull request by cheesinglee**
Here's a more refined version of the calendar adjustment whizzml script that was used in the time series webinar.

Notable differences:
- Now requires a year datestamp as well, with which we perform leap year detection
- Apply calendar adjustment to multiple data fields at once
- Automatic detection of data, month, and year fields

### CODE REVIEW

File: calendar-adjustment/metadata.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "Calendar adjustment for monthly time series",
+  "description": "Given a dataset containing monthly time series data with datestamps, normalizes the time series values by the number of days per month (with leap year detection). Suitable for monthly aggregated data (e.g. total liters per month).",
```
Comment (jaor) [inline calendar-adjustment/metadata.json:3]: adding here a heads-up to @osroca, who will tell us if this description is too long to show nicely in the UI...

File: calendar-adjustment/metadata.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  "kind": "script",
+  "source_code": "script.whizzml",
+  "inputs": [
+    {
+      "name": "id",
+      "type": "dataset-id",
+      "description": "The dataset that contains one or more datestamped monthly time series"
+    },
+    {
+      "name" : "data-fds",
```
Comment (jaor) [inline calendar-adjustment/metadata.json]: i would call this parameter, based on its description, simply `field-ids`

File: calendar-adjustment/metadata.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      "type" : "list",
+      "default" : [],
+      "description" : "List of field IDs which contain unadjusted monthly time series. Defaults to every numeric, parent field."
```
Comment (jaor) [inline calendar-adjustment/metadata.json]: another nit: "parent" is a bit of internal lingo... for users out there, i'd say "non-autogenerated"

File: calendar-adjustment/metadata.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    {
+      "name" : "month-fd",
```
Comment (jaor) [inline calendar-adjustment/metadata.json]: "month-field-id'?

File: calendar-adjustment/metadata.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      "description" : "Field ID for month field. Defaults to first field with datatype = 'month'",
+      "type" : "string",
+      "default" : ""
+    },
+    {
+      "name" : "year-fd",
```
Comment (jaor) [inline calendar-adjustment/metadata.json]: you know what i'm going to say :)

File: readme.md [hunk 6]
```diff
@@ -96,10 +96,10 @@ create all of the example's scripts and libraries for you.
 
 ### Using the web UI
 
-- Install each of the libraries seperately, using the urls to each of
+- Install each of the libraries separately, using the urls to each of
   their folders. (For example, https://github.com/whizzml/examples/tree/master/clean-dataset/clean-data)
 
-- Install each of the scripts seperately, using the
+- Install each of the scripts separately, using the
```
Comment (jaor) [inline readme.md:106]: yes! thanks :) also, somewhere in this readme there's a list with brief descriptions of the provided scripts.  let's add an entry for this one too.

**Review by jaor — CHANGES_REQUESTED** [review]
very minor nits, but oh well, i had to complain of something!

**Comment by jaor** [comment]
@cheesinglee oh, one more thing: it'd be nice if we had a test shell script for this whizzml script

**Comment by cheesinglee** [comment]
nits fixed, readme updated, and test script added.

File: calendar-adjustment/metadata.json (hunk 1 above)
Comment (osroca) [reply calendar-adjustment/metadata.json:3]: thanks for asking but is that easy as to visit the current UI, put the description you want in the script description (More Info > Info > Description), save it and check how it results in the UI. Even one can check how it results if enhanced with some markdown sugar (I recommend to copy the description form the API JSON response once saved with markdown to keep the new lines and all this kind of special things).
image

**Review by jaor — APPROVED** [review]

**Comment by jaor** [comment]
thanks @cheesinglee , everything looks good now.  i took the liberty to rebase it myself before pushing, so closing manually.

**Pull request closed without merging**